packages/react-client/src/ReactFlightReplyClient.js

```diff
@@ -133,33 +139,39 @@ export function processReply(
     value: ReactServerValue,
   ): ReactJSONValue {
     const parent = this;
-    if (__DEV__) {
-      // $FlowFixMe[incompatible-use]
-      const originalValue = this[key];
-      if (typeof originalValue === 'object' && originalValue !== value) {
-        if (objectName(originalValue) !== 'Object') {
-          console.error(
-            'Only plain objects can be passed to Server Functions from the Client. ' +
-              '%s objects are not supported.%s',
-            objectName(originalValue),
-            describeObjectForErrorMessage(parent, key),
-          );
-        } else {
-          console.error(
-            'Only plain objects can be passed to Server Functions from the Client. ' +
-              'Objects with toJSON methods are not supported. Convert it manually ' +
-              'to a simple value before passing it to props.%s',
-            describeObjectForErrorMessage(parent, key),
-          );
+
+    function warnForToJSON() {
```

Comment:
Instead of making this a helper, it's probably fine to special case it not to warn for Dates since that's the only toJSON that's allowed.

Reply:
okey dokey